android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java

```diff
@@ -45,30 +57,51 @@ public static void promptToInstallWallet() {
     if (act == null) {
       return;
     }
+
+    final String url = "market://details?id=" + BuildConfig.BDS_WALLET_PACKAGE_NAME;
+
+    final Intent appStoreIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+
     AlertDialog.Builder builder;
     builder = new AlertDialog.Builder(act);
-    builder.setTitle("APPC Wallet Missing");
-    builder.setMessage("To complete your purchase, you have to install an AppCoins wallet");
-    Log.d("String name: ", "To complete your purchase, you have to install an AppCoins wallet");
+    builder.setTitle(R.string.wallet_missing);
+    builder.setMessage(R.string.install_wallet_from_iab);
+    Log.d("String name: ", act.getString(R.string.install_wallet_from_iab));
 
-    builder.setPositiveButton("INSTALL", new DialogInterface.OnClickListener() {
+    builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialog, int which) {
-        act.startActivity(
-            new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName)));
+        OnInstallClickAction(act, appStoreIntent, url);
       }
     });
 
-    builder.setNegativeButton("SKIP", new DialogInterface.OnClickListener() {
+    builder.setNegativeButton(R.string.skip, new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialogInterface, int i) {
         dialogInterface.cancel();
       }
     });
 
     builder.setIcon(android.R.drawable.ic_dialog_alert);
-    Log.d(TAG, "promptToInstallWallet: builder.show()");
     builder.show();
   }
 
+  private static void OnInstallClickAction(Activity act, Intent appStoreIntent, String url) {
```

Comment:
you should create the AppStoreIntent in one method, here you have it half of it created on promptToInstallWallet and the other half in OnInstallClickAction